Symfony 4

chregu opened the pull request.

chregu committed:
make api_secret optional

chregu committed:
support symfony 4

chregu committed:
test against symfony 4.0

chregu committed:
temporarely use rokka/client-cli symfony-4 branch

dbu committed:
handle command prefix in lazy commands

chregu committed:
Merge pull request #12 from dbu/symfony-4

handle command prefix in lazy commands

dbu committed:
report problem in case of error

dbu committed:
update phpunit tests for symfony 4

dbu committed:
handle prefix correctly

chregu committed:
set branch alias correctly

chregu committed:
require rokka/client-cli 1.4.0

chregu committed:
prepare 1.1.0 release

chregu committed:
styleci

Tobion commented on .travis.yml:
```diff
@@ -40,6 +40,9 @@ matrix:
     - php: 7.1
       env:
         - SYMFONY_VERSION='3.2.*'
```
symfony 3.1 and 3.2 are outdated already and non-lts. so i'd say we don't need to tests against those anymore but instead 3.4 which is lts

chregu replied:
I added 3.4 to the tests (as default for all php versions)
3.2 is officially still security supported for a month. But yes, we could remove those two.

dbu committed:
adjust kernel dir as well, for old symfony versions

Tobion commented on src/DependencyInjection/Configuration.php:
```diff
@@ -29,7 +29,7 @@ public function getConfigTreeBuilder()
                 ->end()
                 ->scalarNode('api_secret')
                     ->info('Api-secret value from user registration.')
```
please update the info if this is not meant to be used anymore.

dbu commented on tests/Functional/Fixtures/AppKernel.php:
```diff
@@ -1,5 +1,7 @@
 <?php
 
+namespace Rokka\RokkaClientBundle\Tests\Functional\Fixtures;
```
@Tobion how does this work with both older and newer symfony tests? or should i make this class without namespace and then make composer dev classmap find it?

Tobion replied:
I'd keep it like this and create a `class_alias` for AppKernel to the namespaced class

chregu committed:
removed api_secret from examples and comment it as deprecated

chregu committed:
test against 3.4 by default

Tobion committed:
fix tests and service visibility

- class alias must be defined after the class is defined
- explicit public declaration on services for compatibility between sf 3 and 4
- use conditional config deprecation for api secret
- better travis according to https://github.com/symfony/symfony-docs/pull/8701

Tobion committed:
sf 3.2.0 fails because it requires annotation, fixed with https://github.com/symfony/symfony/commit/e38be091cef535206cf1d6625806bf5f17095f53

chregu commented:
Thanks a lot for your help everyone
